celery_utils/logged_task.py

```diff
@@ -0,0 +1,34 @@
+"""
+Improved logging for celery tasks.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from celery import Task
+
+
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=abstract-method
+class LoggedTask(Task):
+    """
+    Task base class that emits a log statement when it gets submitted.
+    """
+
+    abstract = True
+
+    def apply_async(self, args, kwargs, **options):
+        """
+        Emit a log statement when the task is submitted.
+        """
+        result = super(LoggedTask, self).apply_async(args, kwargs, **options)
+        log.info('Task {}[{}] submitted with arguments {}, {}'.format(
+            self.name,
+            result.id,
+            args,
+            kwargs
+        ))
```

Comment:
Regarding No 1: It's not _task duration_ that I need.  That, I am able to easily find in Splunk. What I'd like to know is the _task delay_ - how long a celery task has been sitting around before getting run.  This is needed so we can develop an SLA for our grade updates.  So, as a follow up to this, if you are able to write a Splunk query that allows us to compute _task delay_ trends, we're all set.

Regarding No 2: thanks.